projects/project-player-library/src/lib/pages/details-page/details-page.component.ts

```diff
@@ -1,15 +1,175 @@
 import { Component, OnInit } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { Router } from '@angular/router';
+import { DailogPopupComponent } from '../../shared/dialog-popup/dailog-popup.component';
 
 @Component({
   selector: 'lib-details-page',
   templateUrl: './details-page.component.html',
-  styleUrls: ['./details-page.component.css']
+  styleUrls: ['./details-page.component.css'],
 })
 export class DetailsPageComponent implements OnInit {
+  completedCount: number = 0;
+  progressValue: number = 0;
+  projectActions = [
+    {
+      label:"DOWNLOAD",
+      icon1:"cloud_download",
```

Comment:
Change the key name from icon1 to icon
Change the key from emitdata to action
And why is the value of emitData is set to false everywhere?

Reply:
as the emitdata changed to action if the action is true then the data will synced and once the change in data the action is again changed to false.